code/src/components/Submission.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/react-in-jsx-scope */
+import { useState, useEffect } from 'react';
+
+import './Submission.css';
+
+export default function Submission(props) {
+  const [submissionText, setSubmissionText] = useState(
+    localStorage.getItem('localStorageText') || ''
+  );
+
+  useEffect(() => {
+    localStorage.setItem('localStorageText', submissionText);
+  }, [submissionText]);
+
+  function submit(e) {
+    e.preventDefault();
+
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ message: submissionText })
+    })
+      .then((response) => {
+        props.setNeedsUpdating(true);
```

Comment:
Hmm, now I see what was the reason to use `needsUpdating` state property.
It's not the best approach, since it's very complicated. Instead of triggering `useEffect` like this, you could just pass function that has `fetch` request inside into `Submission` component and call it over here